packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.ts

```diff
@@ -0,0 +1,260 @@
+    const amplifyMeta = context.amplify.readJsonFile(amplifyMetaFilePath);
+    const resourceDependsOn = amplifyMeta.function[answers.resourceName].dependsOn || [];
+    answers.dependsOn = resourceDependsOn;
+
+    const { topLevelComment } = await askExecRolePermissionsQuestions(context, answers, newParams, currentDefaults);
+
+    const cfnFileName = `${resourceAnswer.resourceName}-cloudformation-template.json`;
+    const cfnFilePath = path.join(resourceDirPath, cfnFileName);
+    const cfnContent = context.amplify.readJsonFile(cfnFilePath);
+    const dependsOnParams = { env: { Type: 'String' } };
+
+    Object.keys(answers.resourcePropertiesJSON)
+      .filter(resourceProperty => 'Ref' in answers.resourcePropertiesJSON[resourceProperty])
+      .forEach(resourceProperty => {
+        dependsOnParams[answers.resourcePropertiesJSON[resourceProperty].Ref] = {
+          Type: 'String',
+          Default: answers.resourcePropertiesJSON[resourceProperty].Ref,
+        };
+      });
+
+    cfnContent.Parameters = getNewCFNParameters(cfnContent.Parameters, currentParameters, dependsOnParams, newParams);
+
+    Object.assign(answers.resourcePropertiesJSON, { ENV: { Ref: 'env' }, REGION: { Ref: 'AWS::Region' } });
+
+    if (!cfnContent.Resources.AmplifyResourcesPolicy) {
+      cfnContent.Resources.AmplifyResourcesPolicy = {
+        DependsOn: ['LambdaExecutionRole'],
+        Type: 'AWS::IAM::Policy',
+        Properties: {
+          PolicyName: 'amplify-lambda-execution-policy',
+          Roles: [
+            {
+              Ref: 'LambdaExecutionRole',
+            },
+          ],
+          PolicyDocument: {
+            Version: '2012-10-17',
+            Statement: [],
+          },
+        },
+      };
+    }
+
+    if (answers.categoryPolicies.length === 0) {
+      delete cfnContent.Resources.AmplifyResourcesPolicy;
+    } else {
+      cfnContent.Resources.AmplifyResourcesPolicy.Properties.PolicyDocument.Statement = answers.categoryPolicies;
+    }
+
+    cfnContent.Resources.LambdaFunction.Properties.Environment.Variables = getNewCFNEnvVariables(
+      cfnContent.Resources.LambdaFunction.Properties.Environment.Variables,
+      currentParameters,
+      answers.resourcePropertiesJSON,
+      newParams,
+    ); // Need to update
+    // Update top level comment in app.js or index.js file
+
+    const updateTopLevelComment = filePath => {
+      const commentRegex = /\/\* Amplify Params - DO NOT EDIT[a-zA-Z0-9\-\s._=]+Amplify Params - DO NOT EDIT \*\//;
+      let fileContents = fs.readFileSync(filePath).toString();
+      const commentMatches = fileContents.match(commentRegex);
+      if (!commentMatches || commentMatches.length === 0) {
+        fileContents = topLevelComment + fileContents;
+      } else {
+        fileContents = fileContents.replace(commentRegex, topLevelComment);
+      }
+      fs.writeFileSync(filePath, fileContents);
+    };
+    const appJSFilePath = path.join(resourceDirPath, 'src', 'app.js');
```

Comment:
I agree. However, since this PR is already huge and I think lifecycle hooks like that would require some significant rework, I think we should leave update as-is to reduce the blast radius of this PR.